src/server/network.ts

```diff
@@ -263,6 +264,24 @@ export type ResourceTiming = {
   responseStart: number;
 };
 
+export type IPAddressAndPort = {
+  ipAddress?: string;
+  port?: number;
+}
+
+export type SecurityDetails = {
+    protocol?: string;
+    subjectName?: string;
+    issuer?: string;
+    validFrom?: number;
+    validTo?: number;
+};
+
+export type ConnectionDetails = {
```

Comment:
serverIPAddressAndPort and securityDetails seem to be largely independent from one another, and being merged together in this struct appears to complicate the code in this file. Maybe they are two separate top level items?